jung-algorithms/src/test/java/edu/uci/ics/jung/algorithms/shortestpath/TestMinimumSpanningTree.java

```diff
@@ -19,6 +19,7 @@ public void testSimpleTree() {
 
     assertEquals(tree.nodes(), mst.nodes());
     assertEquals(tree.edges(), mst.edges());
+    assertEquals(tree.asGraph(), mst.asGraph());
```

Comment:
As the primary author of the Network API, I'd like to check in with you regarding 
(a) what you believe that the asGraph() method does 
(b) what value you believe that this call adds to the two assertEquals() calls above, and vice versa
(whether or not you've read the Javadoc, although it would be useful to know which).

I will respond further on this once I've heard back.  I don't mean to be mysterious; I just don't want to prejudice your responses, and I want to know what I should do to improve the common.graph APIs or their documentation.

I want to clarify: I'm sure that this does actually catch a bug (thanks!), and I know why, but the way that you've fixed this suggests that there's something about the Network API that isn't clear, and I want to address that.

Reply:
This was puzzling to me at first because I just grabbed my assertEquals from my previous test:

public void testSanity() {
    MutableNetwork<String, Integer> network = NetworkBuilder.directed().build();
    network.addNode("ROOT");
    network.addEdge("ROOT", "LEAF0", 0);
    network.addEdge("ROOT", "LEAF1", 1);

    Network<String, Integer> mst = MinimumSpanningTree.extractFrom(network, e -> 1.0);
    Assert.assertEquals(network, mst);
  }

That test compared 2 MutableNetworks that should be the same.

When I only moved the assertEquals into an existing test, I found that I was asserting that a 
MutableCTreeNetwork was equal to a MutableNetwork (and it was failing I think on the instanceof check). I don't recall, but perhaps they do not (and maybe should not) implement equals in a way that would make this assertion pass.

I took an easy path and just called asGraph() on both sides to pass the assertion.

I think there could be better unit tests, but I wanted to at least leave in anything that was useful during my testing, and that could raise a flag if something gets broken in the future.

Reply:
Fundamentally what's going on here is that I screwed up: `DelegateCTreeNetwork` (and the other tree impls) need to implement `hashCode()` and `equals()` because of course they do.  Filed #132 to fix this.

I mean, I'd be curious to confirm exactly how that error manifested on your side, but regardless that's a bug.

Once that's fixed we should make this just compare the `Network`s directly; that will actually do the right thing.  (At the moment we are comparing only how the nodes are connected, not whether the correct edge objects are being used to connect them.)

Reply:
To clarify: a `CTreeNetwork` inherits the same contract for `equals()` that the common.graph `Network` implementations do (that is, it's defined in terms of the structure, not in terms of the capabilities).   So we should definitely implement `equals()` in such a way these two things should compare as equal.